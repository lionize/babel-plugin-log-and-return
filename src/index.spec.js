import stripIndent from 'strip-indent'
import * as recast from 'recast'
import * as babel from 'babel-core'
import template from 'babel-template'
import plugin from '.'

test('transpiles @log declarator into logAndReturn calls', () => {
  const source = stripIndent(
    `
      // @log
      a + b

      // @log
      hello('world')

      function test() {
        // @log
        return a + b + c
      }
    `,
  ).trim()
  const code = transpile(source)
  expect(code).toMatchSnapshot()
})

function transpile(source) {
  const { code } = babel.transform(source, {
    parserOpts: { parser: recast.parse, plugins: 'template' },
    generatorOpts: { generator: recast.print, lineTerminator: '\n' },
    babelrc: false,
    plugins: [plugin],
  })
  return code
}
