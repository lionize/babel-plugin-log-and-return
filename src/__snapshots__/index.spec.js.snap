// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transpiles @log declarator into logAndReturn calls 1`] = `
"_l(a + b);
_l(hello('world'));

function test() {
  return _l(a + b + c);
}

function _l(x) {
  console.log(x);
  return x;
}"
`;
